app/Http/Controllers/CarController.php

```diff
@@ -38,11 +40,12 @@ public function store(Request $request)
          //validation
          $validatedData = $request->validate([
             'carTitle' => 'required|string',
+            'category_id' => 'required',
             'image' => 'nullable|image|mimes:jpeg,png,jpg,gif,svg|max:2048',
             'discription' => 'required',
             'price' => 'required',
         ]);
-        $image_name= !isset($request->image)?"null":$this->upload_file($request->image ,'assets/images');
+        $image_name= !isset($request->image)?"null":$this->upload_file($request->image ,'assets/images/car');
```

Comment:
modify this part it is hard to read it, you can change image from`nullable` to `required` so the user have to upload image in create

Reply:
is there another way to keep it nullable & make this line readable. in case of the image was optional ?

Reply:
for sure😊
convert it to if condition, take this hint when you find yourself write long `?:` conditional operator convert it to if condition

updates in migration
```php 
$table->string('image')->nullable();
```

updates in controller
```php
if ($request->hasFile('image')) {
    // upload the image
    $data['image'] = $image_name;
}
```